test/component/custodian/state-shift.test.js

```diff
@@ -37,99 +32,43 @@ test('custodian state shall shifts', t => {
     }
   }
   var custodian = new Custodian(runtime)
-  t.false(custodian.isRegistering())
-  t.false(custodian.isPrepared())
-  t.true(custodian.isNetworkUnavailable())
-
-  custodian.onNetworkConnect()
-  t.true(custodian.isRegistering(), 'custodian shall be network connected')
-  t.false(custodian.isPrepared())
-  t.false(custodian.isNetworkUnavailable())
+  custodian.networkAgent.deinit()
 
-  custodian.onLoggedIn()
-  t.false(custodian.isRegistering())
-  t.true(custodian.isPrepared(), 'custodian shall be logged in')
-  t.true(custodian.isLoggedIn(), 'custodian shall be logged in')
-  t.false(custodian.isNetworkUnavailable())
+  /* Replace flora with NetworkMock */
+  custodian.networkAgent._flora = new NetworkMock()
 
-  custodian.onNetworkDisconnect()
-  t.false(custodian.isRegistering())
-  t.false(custodian.isPrepared())
-  t.true(custodian.isLoggedIn(), 'custodian shall be still logged in')
-  t.true(custodian.isNetworkUnavailable(), 'custodian shall be network unavailable on network disconnected')
-})
+  /* Mock bluetooth */
+  custodian.initBluetooth()
+  custodian.bluetoothStream.disconnect()
+  custodian.openBluetooth = () => {}
+  custodian.closeBluetooth = () => {}
 
-test('custodian shall start network app on network disconnect if not logged in', t => {
-  t.plan(4)
-  mock.mockReturns(wifi, 'getNumOfHistory', 0)
+  /* test start */
+  t.true(custodian.isUnconfigured())
+  custodian.configureNetwork()
+  t.true(custodian.isConfiguringNetwork())
 
-  var runtime = {
-    resetNetwork: function () {
-      t.pass('onNetworkDisconnect shall trigger runtime#resetNetwork')
-    },
-    component: {
-      appScheduler: {
-        appMap: {}
-      },
-      lifetime: {
-        getCurrentAppId: () => undefined
-      },
-      light: {
-        stop: function () {}
-      },
-      wormhole: {
-        setOffline: function () {
-          t.fail('onNetworkDisconnect shall not trigger runtime.wormhole#setOffline')
-        }
-      }
+  custodian.bluetoothStream.emit('data', {
```

Comment:
custodian should not directly talks to bluetooth streams as designed. Instead how ssid/password should be retrieved should be delegated to @yoda/network app.